src/pages/studyView/table/StructuralVariantMultiSelectionTable.tsx

```diff
@@ -0,0 +1,782 @@
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+        if (!this.modalSettings.modalOpen) {
+            return;
+        }
+        this.modalSettings.modalPanelName = panelName;
+    }
+
+    @action.bound
+    closeModal() {
+        this.modalSettings.modalOpen = !this.modalSettings.modalOpen;
+    }
+
+    @autobind
+    toggleCancerGeneFilter(event: any) {
+        event.stopPropagation();
+        this.props.onChangeCancerGeneFilter(!this.props.filterByCancerGenes);
+    }
+
+    @computed get isFilteredByCancerGeneList() {
+        return (
+            !!this.props.cancerGeneFilterEnabled &&
+            this.props.filterByCancerGenes
+        );
+    }
+
+    @computed get allSelectedRowsKeysSet() {
+        return stringListToSet([
+            ...this.props.selectedRowsKeys,
+            ...this.preSelectedRowsKeys,
+        ]);
+    }
+
+    @autobind
+    isChecked(uniqueKey: string) {
+        return !!this.allSelectedRowsKeysSet[uniqueKey];
+    }
+
+    @autobind
+    isDisabled(uniqueKey: string) {
+        return _.some(this.preSelectedRowsKeys, key => key === uniqueKey);
+    }
+
+    @action.bound
+    onStructVarHover(rowId: string, isHovered: boolean) {
+        if (isHovered) {
+            this.hoveredStructVarTableRowIds.push(rowId);
+        } else {
+            _.pull(this.hoveredStructVarTableRowIds, rowId);
+        }
+    }
+
+    @action.bound
+    toggleSelectRow(uniqueKey: string) {
+        const record = _.find(
+            this.props.selectedRowsKeys,
+            key => key === uniqueKey
+        );
+        if (_.isUndefined(record)) {
+            this.props.onChangeSelectedRows(
+                this.props.selectedRowsKeys.concat([uniqueKey])
+            );
+        } else {
+            this.props.onChangeSelectedRows(
+                _.xorBy(this.props.selectedRowsKeys, [record])
+            );
+        }
+    }
+    @observable private _selectionType: SelectionOperatorEnum;
+
+    @action.bound
+    afterSelectingRows() {
+        if (this.selectionType === SelectionOperatorEnum.UNION) {
+            this.props.onSubmitSelection([this.props.selectedRowsKeys]);
+        } else {
+            this.props.onSubmitSelection(
+                this.props.selectedRowsKeys.map(selectedRowsKey => [
+                    selectedRowsKey,
+                ])
+            );
+        }
+        this.props.onChangeSelectedRows([]);
+    }
+
+    @computed get selectionType() {
+        if (this._selectionType) {
+            return this._selectionType;
+        }
+        switch (
+            (localStorage.getItem(this.props.tableType) || '').toUpperCase()
+        ) {
+            case SelectionOperatorEnum.INTERSECTION:
+                return SelectionOperatorEnum.INTERSECTION;
+            case SelectionOperatorEnum.UNION:
+                return SelectionOperatorEnum.UNION;
+            default:
+                return this.props.tableType === FreqColumnTypeEnum.DATA
```

Comment:
maybe comment to explain why we default in this way?

Reply:
This code is again directly copied from MultiSelectionTable.tsx.